src/net/sourceforge/kolmafia/CoinmasterData.java

```diff
@@ -6,119 +6,662 @@
+   * associated with it. These are unique across all shops, even if they refer to the same item.
+   *
+   * <p>If the rows are not (all) specified in <code>coinmasters.txt</code> - usually because they
+   * are variable or seasonal (see Mr. Store or The Swagger Shop), this is how you provide a
+   * runtime-constructed map from itemId -> row #.
+   *
+   * @param itemRows - A map from itemId -> row # for shop inventory
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withItemRows(Map<Integer, Integer> itemRows) {
     this.itemRows = itemRows;
+    return this;
+  }
+
+  /**
+   * Provides the base URL for purchasing from this coinmaster. It will be augmented with additional
+   * fields for action, item, and quantity
+   *
+   * @param buyURL - The URL for purchasing from this coinmaster
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withBuyURL(String buyURL) {
     this.buyURL = buyURL;
+    return this;
+  }
+
+  /**
+   * Provides the value of the "action" field specifying that you are buying an item from this
+   * coinmaster.
+   *
+   * @param buyAction - The "action" field specifying a purchase
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withBuyAction(String buyAction) {
     this.buyAction = buyAction;
+    return this;
+  }
+
+  /**
+   * Provides a <code>List</code> of <code>AdventureResult</code>s of the items that you can buy
+   * from this coinmaster.
+   *
+   * <p>If the items are all provided in <code>coinmasters.txt</code>, pass in the "master" name to
+   * get all the items for that coinmaster.
+   *
+   * @param master - The name of the shop in <code>coinmasters.txt</code>
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withBuyItems(String master) {
+    return withBuyItems(CoinmastersDatabase.getBuyItems(master));
+  }
+
+  /**
+   * Provides an empty <code>List</code> of <code>AdventureResult</code>s of the items that you can
+   * buy from this coinmaster.
+   *
+   * <p>The <code>List</code> is expected to be filled in programmatically at runtime when you visit
+   * the coinmaster and see what is for sale.
+   *
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withBuyItems() {
+    return withBuyItems(CoinmastersDatabase.getNewList());
+  }
+
+  /**
+   * Provides a <code>List</code> of <code>AdventureResult</code>s of the items that you can buy
+   * from this coinmaster.
+   *
+   * <p>If not all of the items are in <code>coinmasters.txt</code> - or a <code>List</code> has
+   * been created ahead of time elsewhere - pass it in here.
+   *
+   * <p>*** It is important that the <code>List</code> be a <code>LockableListModel</code>, since it
+   * will be used as the model for this coinmaster in <code>CoinmastersFrame</code> (I need to
+   * figure out how to enforce this, since that will not be the case if we are running headless)
```

Comment:
if we're running headless there is no frame so it doesn't matter, right?

Reply:
Yes. Except our convention seems to be this:
```
  List<AdventureResult> activeEffects = LockableListFactory.getInstance(AdventureResult.class);
```
rather than this, from CoinmastersDatabase:
```
  public static final LockableListModel<AdventureResult> getNewList() {
    return new LockableListModel<AdventureResult>();
  }
```
I want to study this more and, perhaps, deal with it in a subsequent PR.